File: euler/haskell_solutions/four.hs
-- Project Euler Problem Number Four:
--
-- Find the largest palindrome made from
-- the product of two three-digit numbers.

problemFour :: Integer
problemFour = maximum [x | y <- [100..999], z <- [y..999], let x = y * z, let s = show x, s == reverse s]

-- problemFour ==> 906609

File: euler/clojure_solutions/four.clj
; Project Euler Problem Number Four:
;
; Find the largest palindrome made from
; the product of two three-digit numbers.

(defn palindrome? [string]
  (= (reverse string) (seq string)))

(defn problemFour []
  (apply max (filter #(palindrome? (str %)) (for [x (range 100 999) y (range 100 999)] (* x y)))))

; (problemFour) ==> 906609

